vaadin-notification-flow-parent/vaadin-notification-flow-integration-tests/src/main/java/com/vaadin/flow/component/notification/tests/NotificationView.java

```diff
@@ -76,7 +76,7 @@ private void createNotificationUsingStaticConvenienceMethod() {
         Notification notification = Notification.show(
                 "This is a notification created with static convenience method");
         notification.setId("static-notification");
-        addCard("Notification using static convenience method", notification);
```

Comment:
The test setup here was strange (yet supported): the notification was first auto-added to UI using `Notification.show()` but then the same instance got manually added to the card.

The assertion in the related test however was wrong. It assumed that the instance would automatically get removed from its parent on close even though it's manually re-added in the setup. This used to work because the [order of checking the condition and beforeClientResponse was kind of backwards](https://github.com/vaadin/flow-components/blob/5f95ad21b58855d71287cb1cefa716ab5f1e3ef6/vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java#L436-L437) (this PR fixes the order & adds a related test).